django/contrib/auth/models.py

```diff
@@ -162,24 +163,36 @@ def create_superuser(self, username, email, password, **extra_fields):
 
 
 # A few helper functions for common logic between User and AnonymousUser.
-def _user_get_all_permissions(user, obj):
+def _user_get_all_permissions(user, obj, fallback_to_model=None):
     permissions = set()
     for backend in auth.get_backends():
         if hasattr(backend, "get_all_permissions"):
-            permissions.update(backend.get_all_permissions(user, obj))
+            try:
+                permissions.update(backend.get_all_permissions(
+                    user, obj, fallback_to_model))
+            except TypeError:
```

Comment:
This seems overly simplistic, we should at least be sure to the extend that we should check the signature of the function on whether or not it supports more than two arguments.

Reply:
To my knowledge, Python prefers simplicity. Can you think of a case that an explicit check would perform better?

Reply:
Any case where the method raises a `TypeError` which is not caused by the missing argument

Reply:
working on an update

Reply:
pushed the update